number/src/lib.rs

```diff
@@ -31,6 +33,60 @@ pub fn log2_exact(n: BigUint) -> Option<usize> {
         .filter(|zeros| n == (BigUint::from(1u32) << zeros))
 }
 
+pub type Columns<F> = Vec<(String, Vec<F>)>;
+
+#[derive(Serialize, Deserialize)]
+/// Like Columns, but each column can exist in multiple sizes
+pub struct VariablySizedColumns<F> {
+    /// Maps each column name to a (size -> values) map
+    columns: Vec<(String, BTreeMap<usize, Vec<F>>)>,
```

Comment:
I don't think we gain much hiding variably sized columns in a collection of them.

Wouldn't it be better to use a type for ONE column that has a variable size? Then there could be free functions that reduce a collection of those to a unique size.

The `len()` function is very misleading for example. Does it return the number of different sizes? If this is just a `Vec<String, VariableLengthColumn<T>>` it's much clearer.

Reply:
done